lib/bloc/backup/backup_bloc.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:async';
+import 'package:breez/services/injector.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:breez/services/breezlib/breez_bridge.dart';
+import 'package:breez/services/breezlib/data/rpc.pb.dart';
+import 'package:breez/services/backup.dart';
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:path_provider/path_provider.dart';
+
+class BackupBloc {
+  BackupService _service;
+  AccountBloc _accountBloc;
+  String _currentNodeId;
+  List<String> _currentBackupPaths;
+
+  final _promptEnableBackupController = new StreamController<bool>.broadcast();
+  Stream<bool> get promptEnableStream => _promptEnableBackupController.stream;
+
+  final _backupDisabledIndicatorController = new StreamController<bool>.broadcast();
+  Stream<bool> get backupDisabledIndicatorStream => _backupDisabledIndicatorController.stream;
+
+  final _disableBackupPromptController = new StreamController<bool>();
+  Sink<bool> get disableBackupPromptSink => _disableBackupPromptController.sink;
+
+  final _backupNowController = new StreamController<bool>();
+  Sink<bool> get backupNowSink => _backupNowController.sink;
+
+  final _restoreRequestController = new StreamController<String>();
+  Sink<String> get restoreRequestSink => _restoreRequestController.sink;
+
+  final _multipleRestoreController = new StreamController<Map<String, String>>.broadcast();
+  Stream<Map<String, String>> get multipleRestoreStream => _multipleRestoreController.stream;
+
+  static const String BACKUP_PROMPT_DISABLED_PREFERENCES_KEY = "backupDisabled";
+
+  BackupBloc(this._accountBloc) {
+    ServiceInjector injector = new ServiceInjector();
+    BreezBridge breezLib = injector.breezBridge;
+    _service = BackupService();
+
+    var sharedPrefrences = SharedPreferences.getInstance();
+
+    _listenBackupPaths(breezLib, sharedPrefrences);
+    _listenEnableBackupRequests(sharedPrefrences);
+    _listenBackupNowRequests(sharedPrefrences);
+    _listenRestoreRequests(breezLib);
+
+    _accountBloc.accountStream.listen((acc) {
+      _currentNodeId = acc.id;
+    });
+  }
+
+  void _listenEnableBackupRequests(Future<SharedPreferences> sharedPrefrences) {
+    _disableBackupPromptController.stream.listen((disable) {
+      sharedPrefrences.then((preferences) {
+        preferences.setBool(BACKUP_PROMPT_DISABLED_PREFERENCES_KEY, disable);
+      });
+    });
+  }
+
+  void _listenBackupNowRequests(Future<SharedPreferences> sharedPrefrences) {
+    _backupNowController.stream.listen((data) {
+      backup();
+    });
+  }
+
+  _listenBackupPaths(BreezBridge breezLib, Future<SharedPreferences> sharedPrefrences) {
+    Observable(breezLib.notificationStream).where((event) {
+      return event.type ==
+          NotificationEvent_NotificationType.BACKUP_FILES_AVAILABLE;
+    }).listen((event) {
+      _currentBackupPaths = event.data;
```

Comment:
Better to pas the backup paths directly to the backup function instead of keeping them in a class variable. It eliminates all sorts of races.